libkbfs/root_metadata.go

```diff
@@ -854,6 +854,42 @@ func (md *RootMetadata) GetHistoricTLFCryptKey(
 		crypto, keyGen, currentKey, md.extra)
 }
 
+// IsWriter checks that the given user is a valid writer of the TLF
+// right now.  Implements the KeyMetadata interface for RootMetadata.
+func (md *RootMetadata) IsWriter(
+	ctx context.Context, checker TeamMembershipChecker, uid keybase1.UID) (
+	bool, error) {
+	h := md.GetTlfHandle()
+	if h.Type() != tlf.SingleTeam {
+		return h.IsWriter(uid), nil
+	}
+
+	// Team membership needs to be checked with the service.
```

Comment:
Perhaps add some thing like "A team is always a single writer from KBFS perspective" as comment?